packages/common/src/store/pages/chat/selectors.ts

```diff
@@ -150,3 +151,62 @@ export const getUnfurlMetadata = (
   const message = getChatMessageById(state, chatId, messageId)
   return message?.unfurlMetadata
 }
+
+export const getUserChatPermissions = createSelector(
+  [getChatPermissions, getUserId],
+  (permissions, userId) => {
+    return userId ? permissions[userId] : undefined
+  }
+)
+
+export const getCanChat = createSelector(
+  [
+    getUserId,
+    getBlockees,
+    getBlockers,
+    getChatPermissions,
+    (_state: CommonState, userId: ID) => {
+      return userId
+    }
+  ],
+  (
+    currentUserId,
+    blockees,
+    blockers,
+    chatPermissions,
+    userId
+  ): { canChat: boolean; callToAction: ChatPermissionAction } => {
+    const currentUserPermissions = currentUserId
+      ? chatPermissions[currentUserId]
+      : null
+    const userPermissions = chatPermissions[userId]
+    const isBlockee = blockees.includes(userId)
+    const isBlocker = blockers.includes(userId)
+    const canChat =
+      !isBlockee &&
+      !isBlocker &&
+      (userPermissions?.current_user_has_permission ?? true)
```

Comment:
this wont be false bidirectionally yet, so the logic below that calcs actions based on the `currentUserPermissions` gated with (`!canChat)` wont be evaluated if the non-current user is not a blocker/blockee and has permissions such that the current user is allowed to chat with them. 

tldr we need a bidirectional `current_user_has_permission` for this to work, and we can calc the action on the backend or the frontend but would prefer to just keep it all contained in the backend (and frontend reads a `to_enable` field in the resp)

Reply:
ahh yeah ok that checks out. Thanks!

Reply:
@michellebrier / @dharit-tan does this work now that we have unidirectional permissions?

Reply:
Think so